test_regress/t/t_uvm_todo.pl

```diff
@@ -14,6 +14,7 @@
     v_flags2 => ["--timing",
                  "-Wno-PKGNODECL -Wno-IMPLICITSTATIC -Wno-CONSTRAINTIGN -Wno-MISINDENT",
                  "-Wno-CASEINCOMPLETE -Wno-CASTCONST -Wno-SYMRSVDWORD -Wno-WIDTHEXPAND -Wno-WIDTHTRUNC",
+                 "-Wno-EVENTEXPR",
```

Comment:
Are you certain that what UVM does is outside of IEEE and so needs this warning?

Reply:
Here's a fragment of `uvm_objection` implementation:
```systemverilog
  virtual function void raised (uvm_object obj,
                                uvm_object source_obj,
                                string description,
                                int count);
    uvm_component comp;
    if ($cast(comp,obj))    
      comp.raised(this, source_obj, description, count);
    `uvm_do_callbacks(uvm_objection,uvm_objection_callback,raised(this,obj,source_obj,description,count))
    if (m_events.exists(obj))
       ->m_events[obj].raised;
  endfunction
```

And here's how the spec defines the grammar for event triggers:

```
event_trigger ::=
  -> hierarchical_event_identifier ;
  | ->> [ delay_or_event_control ] hierarchical_event_identifier ;
```
Let's look at `hierarchical_event_identifier`:
```
hierarchical_event_identifier ::= hierarchical_identifier
```
Ok, so:
```
hierarchical_identifier ::= [ $root . ] { identifier constant_bit_select . } identifier
```
Let's check `constant_bit_select` just to be sure:
```
constant_bit_select ::= { [ constant_expression ] }
```
Ok, the `obj` in
```
       ->m_events[obj].raised;
```
does not look like a `constant_expression` to me.

-----------------

Funnily enough, the `@` operator has a definition that allows for using event expressions:

```
event_control ::=
  @ hierarchical_event_identifier
  | @ ( event_expression )
  | @*
  | @ (*)
  | @ ps_or_hierarchical_sequence_identifier
```

So to me, this looks like an oversight in the spec, especially given that it can be easily worked around by creating an event and assigning a result of an expression to it:

```
       event e_temp = m_events[obj].raised;
       ->e_temp;
```
However, regardless of whether it is an oversight or a deliberate decision, it is a part of the spec.

Reply:
I find it a bit odd to have a warning for such a detail, but not for the fact that _event merging_ is unsupported.

Reply:
Probably just doing the normal UNSUPPORTED here instead of a new EVENTEXPR message is sufficient.

Reply:
@kbieganski I kinda agree, but the problem is that we don't really know whether somebody intends to use that or not. We could throw an extra warning for every assignment that is not done as an initialization. According to a comment in Verilator's sources, even commercial simulators struggled with that.

@wsnyder it is supported by Verilator though and it's required to run verilate UVM. With an extra error code, I can allow Verilator to ignore this arbitrary restriction that comes from the spec by passing `Wno-EVENTEXPR`.

Reply:
Perhaps I misunderstood earlier. If you are saying the warning is just to say this is something supported over an extension over IEEE, we don't need a warning at all.

BTW if we do add warnings they need to be documented in docs/guide/warnings.rst.

Reply:
Ok, I removed this warning.